src/Symfony/Component/Notifier/Bridge/LineBot/LineBotTransportFactory.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\LineBot;
+
+use Symfony\Component\Notifier\Exception\IncompleteDsnException;
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+
+/**
+ * @author Yi-Jyun Pan <me@pan93.com>
+ */
+final class LineBotTransportFactory extends AbstractTransportFactory
+{
+    private const SCHEME = 'linebot';
+
+    protected function getSupportedSchemes(): array
+    {
+        return [self::SCHEME];
+    }
+
+    public function create(Dsn $dsn): LineBotTransport
+    {
+        if (self::SCHEME !== $dsn->getScheme()) {
+            throw new UnsupportedSchemeException($dsn, self::SCHEME, $this->getSupportedSchemes());
+        }
+
+        $accessToken = $dsn->getUser();
+        if (null === $accessToken) {
+            throw new IncompleteDsnException('Missing channel access token.', 'linebot://'.$dsn->getHost());
+        }
+
+        $receiver = $dsn->getRequiredOption('receiver');
+        \assert(\is_string($receiver));
```

Comment:
I think all values are strings, do we need this check?

Reply:
Well, since `getRequiredOption` returns `mixed`, I’m afraid it's not ideal to assume that every option is a string. I can open another issue to discuss whether we can improve the type hint of Dsn, for example, by returning `?string`.

https://github.com/symfony/symfony/blob/d83167dab489e6920e22156939cbd8abdddc92fe/src/Symfony/Component/Notifier/Transport/Dsn.php#L82-L94

Reply:
Let's throw an exception instead? (as we're not using assert anywhere else in the code base).